docs/source/reference.md

```diff
@@ -0,0 +1,134 @@
+# API Reference
+
```

Comment:
I'm a bit worried about this page as it's not updated automatically, i.e., it will likely bitrot when the library gets extended, but I have no better suggestion here at the moment.